### DISCUSSION TITLE
Hold button to change values in options faster

### DISCUSSION THREAD

**Discussion by lovelydumpling** [ikemen-engine/Ikemen-GO#976] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

In the options page in IKEMEN, there are certain options like changing volume or match timer, where the options change only by 1 value per press of the button. If you want to change the timer from 99 to None, you have to press Left 100 times. Same to change it back again.

### Describe the solution you'd like

I'd like to be able to hold the left button down for it to tick through the numbers quickly, and tap to do so one at a time. Perhaps it'll go faster the longer you hold the button, and then you'd stop when you hit near the number you want, and then tap it for finer control. This is how most menus work in my experience. 

### Search terms

menu value tick ticks option options volume timer control "one at a time" "1 at a time"

**Comment by lovelydumpling** [comment]
![image](https://user-images.githubusercontent.com/30986053/220767316-91415be9-ff4e-4b91-8d59-eba7dc27713c.png)
Extension on this, character select could use a similar feature as brought up unprompted by a friend of mine. Large rosters could benefit from being able to hold the button down to navigate instead of having to tap every time.

**Comment by Lazin3ss** [comment]
Some ideas I have for this are:
- Setting an "autoscroll" delay (like, having to wait a certain time after a holding a direction for the game to start autoscrolling some option)
- Setting an autoscroll speed (self-explanatory, I guess)

These could be system.def parameters for diverse options, such as select screen grid, options menu items, etc.

Some naming ideas:
```ini
menu.autoscroll.delay = 60                      ; Default delay for an option (or the menu itself) to start autoscrolling (in ticks)
menu.autoscroll.speed = 10                      ; Default speed at which the autoscroll will change an option value (or menu item) (in ticks) 
menu.itemname.difficulty.autoscroll.delay = 0
menu.itemname.difficulty.autoscroll.speed = 10
```

**Reply by potsmugen** [reply]
Could use rate instead of speed, like the Windows keyboard options (which use delay as well). And maybe be a setting in config.json for all menu items instead of case by case. I just checked a couple of Capcom games and they apparently have the same autoscroll type for all menu items.

**Comment by K4thos** [comment, marked as answer]
implemented in #3333
